app/scripts/metamask-controller.js

```diff
@@ -1093,13 +1095,39 @@ export default class MetamaskController extends EventEmitter {
     });
     this.memStore.subscribe(this.sendUpdate.bind(this));
 
-    const password = process.env.CONF?.PASSWORD;
-    if (
-      password &&
-      !this.isUnlocked() &&
-      this.onboardingController.store.getState().completedOnboarding
-    ) {
-      this.submitPassword(password);
+    if (!this.isUnlocked()) {
+      // Automatic login via config password
+      const password = process.env.CONF?.PASSWORD;
+      if (
+        password &&
+        this.onboardingController.store.getState().completedOnboarding
+      ) {
+        this.submitPassword(password);
+      }
+      // Automatic login via storage encryption key
+      else if (isManifestV3) {
```

Comment:
this says password has to wait for onboarding to be complete, but mv3 login doesn't. I hope that's right.

Reply:
Good catch, will fix!